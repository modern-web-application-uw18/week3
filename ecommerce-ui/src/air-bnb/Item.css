* {
    box-sizing: border-box;
}

.Item {
    display: flex;
    flex-wrap: nowrap;
    height: 50px;
    max-height: 50px;
    width: 1050px;
    margin-left: 10px;
}

.Item-image-container {
    height: 50px;
    width: 50px;
    margin-left: 15px;
    margin-right: 15px;
    margin-top: 5px;
}

.Item-image-container img {
    width: 50px;
    height: auto;
    overflow: hidden;
}

.Item-title-container {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    color: gray;
    height: 25px;
    width: 600px;
    text-align: left;
    color: green;
}

.Item-cost {
    margin-left: 300px;
    font-weight: 500;
    font-size: 16px;
}

.Item-button-container {
    margin-left: 50px;
}

.Item-green-button {
    background-color: darkgreen;
    border: none;
    color: white;
    padding: 10px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 14px;
    margin: 4px 2px;
    cursor: pointer;
    border-radius: 10px;
    font-weight: 700;
}

