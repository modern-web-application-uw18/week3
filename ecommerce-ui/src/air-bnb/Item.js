import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Location from './Location';
import Payment from './Payment';
import './Item.css';
import './Home.css';

class Item extends Component {

    render() {

        const {
            title,
            image,
            location,
            payment
        } = this.props.item;

        return (
            <div className="Item">
                <div className="Item-image-container" >
                    <img src={image} alt="" />
                </div>
                <div className="Item-title-container">
                    <div className="Home-Title">{title}</div>
                    <Location location={location} />
                </div>
                <div className="Item-cost">${payment.cost} per night</div>
                <div className="Item-button-container">
                    <button onClick={this.props.removeFromCart}
                        className="Item-green-button">Remove</button>
                </div>
            </div>
        );
    }
}

Item.propTypes = {
    title: PropTypes.string.isRequired,
    image: PropTypes.string,
    location: PropTypes.instanceOf(Location),
    payment: PropTypes.instanceOf(Payment)
};

Item.defaultProps = {
    title: 'No title :('
};

export default Item;