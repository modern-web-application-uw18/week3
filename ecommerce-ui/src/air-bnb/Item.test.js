import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Item from './Payment';

describe('Item', () => {
    it('renders without crashing', () => {
        const location = {"city": "London", "country": "England"};
        const payment = {"cost": 180, "description": "Free cancellation"}
        const div = document.createElement('div');
        ReactDOM.render(<Item
            title="Rustic flat in Sodo"
            image="https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80"
            location={location}
            payment={payment} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with cost and description', () => {
        const location = {"city": "Manchester", "country": "England"};
        const payment = {"cost": 130, "description": "Free cancellation"}
        const component = TestRenderer.create(
            <Item title="Quaint old home in sleepy neighborhood"
                image="https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80"
                location={location}
                payment={payment} />);
            expect(component.toJSON()).toMatchSnapshot();
    });
});