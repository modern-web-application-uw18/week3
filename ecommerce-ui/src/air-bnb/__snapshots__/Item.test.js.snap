// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render with cost and description 1`] = `
<div>
  $
  130
   per night 
  <span>
     Â· 
    Free cancellation
  </span>
</div>
`;
